lib/dataLayer.tsx

```diff
@@ -421,53 +475,99 @@ function _buildFormDataObject(form: PublicFormSchemaProperties, values: Response
   const formData = new FormData();
   form.elements = form.elements.filter((element) => !["richText"].includes(element.type));
 
-  form.elements.map((element) => {
-    _handleFormDataType(element, values[element.id], formData);
-  });
+  for (const element of form.elements) {
+    const result = _handleDynamicRowTypeIfNeeded(element, values[element.id]);
+    for (const tuple of result) {
+      formData.append(tuple[0], tuple[1]);
+    }
+  }
+
   formData.append("formID", form.formID);
+
   return formData;
 }
 
-function _handleFormDataType(element: FormElement, value: Response, formData: FormData) {
+function _handleDynamicRowTypeIfNeeded(
+  element: FormElement,
+  value: Response
+): [string, string | Blob][] {
+  if (element.type === "dynamicRow") {
+    if (element.properties.subElements === undefined) return [];
+
+    const subElements = element.properties.subElements.filter(
+      (element) => !["richText"].includes(element.type)
+    );
+
+    const responses = value as Responses[];
+
+    /**
+     * We are creating a new data structure (to be passed to the submit API) from the multiple responses that could have been entered
+     * for the dynamic row component we are currently processing. */
+    return (
+      responses
+        .map((response, responseIndex) => {
+          return subElements
+            .map((subElement, subElementIndex) => {
+              const result = _handleFormDataType(subElement, response[subElementIndex]);
```

Comment:
The _handleFormDataType is receiving the wrong response for the subElement.  The response property for the subElement is the index of the subElement in the unmodified form config array.  By removing the richtext element the array has been shifted and subElement with ID 1 is being passed in response[0].
This could be rewritten like: 
```js
const result = _handleFormDataType(subElement, response[subElement.id]);
```

Reply:
Discovered another issue with your approach but I found a new way to fix it :)